lib/core/src/display/shape/text/cursor.rs

```diff
@@ -100,43 +259,93 @@ lazy_static! {
 /// a WebGL buffer with vertex positions of all cursors.
 #[derive(Debug)]
 pub struct Cursors {
-    pub cursors       : Vec<Cursor>,
-    pub dirty_cursors : HashSet<usize>,
-    pub buffer        : WebGlBuffer,
+    pub cursors : Vec<Cursor>,
+    pub dirty   : bool,
+    pub buffer  : Option<WebGlBuffer>,
```

Comment:
I don't like the hierarchy here - mixing display logic with data logic. However, I believe no change is required here, as after moving to sprite system this will be cleaned up. This is just a note for you Adam, so you see how I think about it.